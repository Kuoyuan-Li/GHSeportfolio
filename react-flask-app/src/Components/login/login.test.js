import React from 'react'
import Login from './login'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

describe ('login page test', () => {
    

    const historyMock = { push: jest.fn() }
    const onMock = jest.fn();

    let wrapper;

    beofreAll(() => {
        global.fetch = jest.fn();
    });

    beforeEach(() => {
        wrapper = shallow(<Login history = {historyMock} onSubmit={onMock}/>); 
    });

    test('redirect to index page when click : Back to Index Page', () => {
        wrapper.find('button.backToIndex').simulate('click');       
        expect(historyMock.push.mock.calls[0]).toEqual(['/']);
    });

    test('Input incomplete login Info, miss username and password' , ()=>{
        const form = wrapper.find('form.loginForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '' },
                    password: { value: '' }
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });

    test('Input incomplete login Info, miss username' , ()=>{
        const form = wrapper.find('form.loginForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '' },
                    password: { value: '1' }
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });

    test('Input incomplete login Info, miss password' , ()=>{
        const form = wrapper.find('form.loginForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '1' },
                    password: { value: '' }
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });


    test('Input complete login Info' , () => {
        const form = wrapper.find('form.loginForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '1' },
                    password: { value: '1' }
                } 
            }
        });       
        expect(wrapper.state('message')).toEqual('');      

    });

    test ('Api fetch success, push to profile',()=>{
        const fakeApiResponse1 = {'validity':'True','message':''}
        const fakeApiResponse2 = {'validity':'False','message':'Invalid username or password'}
        var {getByTestLogin} = render(<Index />)
        var apiFunc = jest.spyOn(global, 'onSubmit').mockImplementationOnce(() => {
            return Promise.resolve({
              json: () => Promise.resolve(fakeApiResponse1)
            })
        })    
        const text = await getByTestId("ptag")
        expect(text).toHaveTextContent(fakeUserResponse['data'])

    })
    
    
    test('renders correctly', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();
      });
});