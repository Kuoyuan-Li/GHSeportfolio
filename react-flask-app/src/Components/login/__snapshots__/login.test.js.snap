// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page test renders correctly 1`] = `
<body
  id="login"
>
  <div
    className="container"
  >
    <div
      class="row"
    >
      <form
        className="loginForm"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div />
        <div
          className="form form1"
        >
          <div
            className="form-group"
          >
            <input
              name="username"
              onChange={[Function]}
              placeholder="Type your user name"
              type="username"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              name="password"
              onChange={[Function]}
              placeholder="Type your password"
              type="password"
              value=""
            />
          </div>
          <button
            class="button button1"
            className="login"
            type="submit"
          >
            Log in
          </button>
        </div>
      </form>
    </div>
    <button
      class="linkButton"
      className="backToIndex"
      onClick={[Function]}
    >
      Back to Index Page
    </button>
  </div>
</body>
`;
